workspaces/src/network/mod.rs

```diff
@@ -0,0 +1,169 @@
+mod account;
+mod result;
+mod sandbox;
+mod server;
+mod testnet;
+
+use std::path::Path;
+
+use async_trait::async_trait;
+
+use near_crypto::{InMemorySigner, KeyType, Signer};
+use near_jsonrpc_client::methods::sandbox_patch_state::RpcSandboxPatchStateRequest;
+use near_primitives::borsh::BorshSerialize;
+use near_primitives::state_record::StateRecord;
+use near_primitives::types::AccountId;
+
+use crate::rpc::client::Client;
+
+pub use crate::network::account::{Account, Contract};
+pub use crate::network::sandbox::Sandbox;
+pub use crate::network::testnet::Testnet;
+
+pub use self::result::{CallExecution, CallExecutionDetails};
+
+const DEV_ACCOUNT_SEED: &str = "testificate";
+
+pub trait NetworkClient {
+    fn client(&self) -> &Client;
+}
+
+pub trait NetworkInfo {
+    /// Name of the network itself
+    fn name(&self) -> String;
+
+    /// Root Account ID of the network. Mainnet has `near`, testnet has `testnet`.
+    fn root_account_id(&self) -> AccountId;
+
+    /// Path to the keystore directory
+    fn keystore_path(&self) -> std::path::PathBuf;
+
+    // TODO: change return type to Url instead of String
+    /// Rpc endpoint to point our client to
+    fn rpc_url(&self) -> String;
+
+    // TODO: not everything has a helper url. maybe make this optional or remove it into a seprate trait
+    /// The helper URL to create top level account out of for certain networks.
+    fn helper_url(&self) -> String;
+}
+
+#[async_trait]
+pub trait TopLevelAccountCreator {
+    async fn create_tla(
+        &self,
+        id: AccountId,
+        signer: InMemorySigner,
+    ) -> anyhow::Result<CallExecution<Account>>;
+
+    async fn create_tla_and_deploy<P: AsRef<Path> + Send + Sync>(
+        &self,
+        id: AccountId,
+        signer: InMemorySigner,
+        wasm: P,
+    ) -> anyhow::Result<CallExecution<Contract>>;
+}
+
+// NOTE: Not all networks/runtimes will have the ability to be able to do dev_deploy.
+// This trait acts as segmented boundary for only specific networks such as sandbox and testnet.
+pub trait AllowDevAccountCreation {}
+
+#[async_trait]
+pub trait DevAccountDeployer {
+    fn dev_generate(&self) -> (AccountId, InMemorySigner);
```

Comment:
I'll make the function `async` for now, but I'll add a separate issue to see which crate to make use of. Probably best to use something like `async-compat` since this library would now allow any runtime to be ran w/ it.